packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -7,96 +7,74 @@ import { Text } from '../../../primitives';
 import { UploadButton } from './UploadButton';
 import { Previewer } from './Previewer';
 import { UploadDropZone } from './UploadDropZone';
+import { Tracker } from './Tracker';
 
 export function FileUploader({
   acceptedFileTypes,
+  components = {},
   fileNames,
   isPreviewerVisible,
   level,
-  components = {},
-  multiple = true,
-  variation = 'button',
-  maxSize,
   maxFiles,
+  maxSize,
+  multiple = true,
   onError,
   onSuccess,
+  variation = 'button',
 }: FileUploaderProps): JSX.Element {
   const {
     UploadDropZone = FileUploader.UploadDropZone,
     UploadButton = FileUploader.UploadButton,
   } = components;
   const {
-    files,
+    addTargetFiles,
+    fileStatuses,
     inDropZone,
     onDragEnter,
     onDragLeave,
     onDragOver,
     onDragStart,
     onDrop,
+    setFileStatuses,
     setShowPreviewer,
-    addTargetFiles,
     showPreviewer,
-    setFiles,
-    fileStatuses,
-    setFileStatuses,
-    setFileSizeErrors,
   } = useFileUploader(maxSize, acceptedFileTypes, multiple);
 
-  const [allFileNames, setAllFileNames] = useState<string[]>([]);
   const fileStatusesRef = useRef<FileStatuses>([]);
-  // File Previewer global states
+
+  // File Previewer loading state
   const [isLoading, setLoading] = useState(false);
```

Comment:
I was unable to move this to a derived state. I could not get the render to work correctly when the loading occurs